integration-tests/helpers.js

```diff
@@ -0,0 +1,116 @@
+'use strict'
+
+const express = require('express')
+const bodyParser = require('body-parser')
+const msgpack = require('msgpack-lite')
+const codec = msgpack.createCodec({ int64: true })
+const EventEmitter = require('events')
+const { fork } = require('child_process')
+const http = require('http')
+
+class FakeAgent extends EventEmitter {
+  constructor (port = 0) {
+    super()
+    const agent = express()
+    agent.use(bodyParser.raw({ limit: Infinity, type: 'application/msgpack' }))
+    agent.use((req, res) => {
+      if (req.body.length === 0) return res.status(200).send()
+      this.emit('message', {
+        headers: req.headers,
+        payload: msgpack.decode(req.body, { codec })
+      })
+    })
+    this.server = http.createServer(agent).listen(port, () => {
+      this.port = this.server.address().port
+      this.emit('listening', this.port)
+    })
+  }
+
+  listeningPort () {
+    return new Promise((resolve) => {
```

Comment:
This could potentially never resolve if the event was already emitted.

Reply:
It also doesn't seem to be used in any case?

Reply:
Correct. It was used in a previous iteration of the tests. Removing.

Reply:
Actually it was supposed to be used. I'm making a better version. Stay tuned.